packager/binarypackaging.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+#
+#     Copyright (C) 2018 Team Kodi
+#     http://kodi.tv
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import zipfile
+from distutils.version import LooseVersion
+from distutils.dir_util import copy_tree
+from itertools import groupby
+from xml.etree import ElementTree as ET
+
+from packager import utils
+from .utils import delete_companion_files, makedirs_ignore_errors, meets_version_requirements, pack_artifact
+
+
+logger = logging.getLogger(__name__)
+
+Artifact = namedtuple('Artifact', ['addon_id', 'version', 'binary_repo', 'platform'])
```

Comment:
'binary_repo' -> 'path' or 'location'. Since it's the path/location of the artifact..